sponsorship/src/main/java/com/kyriba/conference/sponsorship/service/SponsorService.java

```diff
@@ -1,14 +1,56 @@
 package com.kyriba.conference.sponsorship.service;
 
+import com.kyriba.conference.sponsorship.dao.SponsorRepository;
+import com.kyriba.conference.sponsorship.domain.EmailMessage;
+import com.kyriba.conference.sponsorship.domain.Sponsor;
+import com.kyriba.conference.sponsorship.domain.dto.SponsorDto;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
+
+import javax.transaction.Transactional;
 
 
 /**
  * @author M-ASL
  * @since v1.0
  */
 @Service
+@Transactional
+@RequiredArgsConstructor
 public class SponsorService
 {
-  // todo-asl
+  private final SponsorRepository sponsorRepository;
+  private final EmailService emailService;
+
+
+  public Long createSponsor(String name, String email)
+  {
+    Sponsor sponsor = Sponsor.builder()
+        .name(name)
+        .email(email)
+        .build();
+    sendEmailNotification(sponsor);
+    return sponsorRepository.save(sponsor).getId();
+  }
+
+
+  private void sendEmailNotification(Sponsor sponsor)
+  {
+    try {
+      emailService.sendNotification(EmailMessage.createForSponsor(sponsor));
+    }
+    catch (Exception e) {
+      //todo it is workaround, what behavior is expected when the other service is not available?

```

Comment:
Hard to implement exception handling, because initially such communication should be async with messaging.